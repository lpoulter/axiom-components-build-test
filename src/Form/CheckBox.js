import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Validate from '../Validation/Validate';
import ChedioButtox from './ChedioButtox';

export default class CheckBox extends Component {
  static propTypes = {
    /** Checked state of the input */
    checked: PropTypes.bool,
    /** Label that is placed next to the input */
    children: PropTypes.node,
    /** Disables interactions and applies styling */
    disabled: PropTypes.bool,
    /** See Validate[error] */
    error: PropTypes.func,
    /** Partially checked state of the input */
    indeterminate: PropTypes.bool,
    /** Applies styling to indicate the users input was invalid */
    invalid: PropTypes.bool,
    /** Name of the group the input belongs to */
    name: PropTypes.string,
    /** See Validate[patterns] */
    patterns: PropTypes.array,
    /** See Validate[required] */
    required: PropTypes.bool,
    /** Text that is displayed on hover */
    title: PropTypes.string,
  };

  render() {
    const {
      checked,
      children,
      disabled,
      error,
      invalid,
      name,
      indeterminate,
      patterns,
      required,
      ...rest
    } = this.props;

    return (
      <Validate
          error={ error }
          patterns={ patterns }
          required={ required }
          value={ checked }>
        { (isValid) =>
          <ChedioButtox
              { ...rest }
              checked={ !!(checked || indeterminate) }
              className="ax-checkbox"
              disabled={ disabled }
              indeterminate={ indeterminate && !checked }
              inputType="checkbox"
              invalid={ invalid || isValid === false }
              name={ name }>
            { children }
          </ChedioButtox>
        }
      </Validate>
    );
  }
}
