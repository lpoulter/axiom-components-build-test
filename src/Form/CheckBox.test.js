import React from 'react';
import renderer from 'react-test-renderer';
import CheckBox from './CheckBox';

const getComponent = (props = {}) =>
  renderer.create(
    <CheckBox { ...props }>Lorem ipsum</CheckBox>
  );

describe('CheckBox', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with disabled', () => {
    const component = getComponent({ disabled: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with invalid', () => {
    const component = getComponent({ invalid: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with indeterminate', () => {
    const component = getComponent({ indeterminate: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with indeterminate and checked', () => {
    const component = getComponent({ indeterminate: true, checked: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
