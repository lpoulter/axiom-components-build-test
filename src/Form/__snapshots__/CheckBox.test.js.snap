// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox renders with defaultProps 1`] = `
<label
  className="ax-checkbox ax-space--x2"
>
  <input
    checked={false}
    className="ax-checkbox__input"
    type="checkbox"
  />
  <span
    className="ax-checkbox__indicator"
  />
  <span
    className="ax-checkbox__label"
  >
    Lorem ipsum
  </span>
</label>
`;

exports[`CheckBox renders with disabled 1`] = `
<label
  className="ax-checkbox ax-checkbox--disabled ax-space--x2"
>
  <input
    checked={false}
    className="ax-checkbox__input"
    disabled={true}
    type="checkbox"
  />
  <span
    className="ax-checkbox__indicator"
  />
  <span
    className="ax-checkbox__label"
  >
    Lorem ipsum
  </span>
</label>
`;

exports[`CheckBox renders with indeterminate 1`] = `
<label
  className="ax-checkbox ax-checkbox--indeterminate ax-space--x2"
>
  <input
    checked={true}
    className="ax-checkbox__input"
    type="checkbox"
  />
  <span
    className="ax-checkbox__indicator"
  />
  <span
    className="ax-checkbox__label"
  >
    Lorem ipsum
  </span>
</label>
`;

exports[`CheckBox renders with indeterminate and checked 1`] = `
<label
  className="ax-checkbox ax-space--x2"
>
  <input
    checked={true}
    className="ax-checkbox__input"
    type="checkbox"
  />
  <span
    className="ax-checkbox__indicator"
  />
  <span
    className="ax-checkbox__label"
  >
    Lorem ipsum
  </span>
</label>
`;

exports[`CheckBox renders with invalid 1`] = `
<label
  className="ax-checkbox ax-checkbox--invalid ax-space--x2"
>
  <input
    checked={false}
    className="ax-checkbox__input"
    type="checkbox"
  />
  <span
    className="ax-checkbox__indicator"
  />
  <span
    className="ax-checkbox__label"
  >
    Lorem ipsum
  </span>
</label>
`;
